pkg/broker/deprovision_subscriber.go

```diff
@@ -36,64 +35,40 @@ func NewDeprovisionWorkSubscriber(dao *dao.Dao) *DeprovisionWorkSubscriber {
 func (d *DeprovisionWorkSubscriber) Subscribe(msgBuffer <-chan JobMsg) {
 	d.msgBuffer = msgBuffer
-
 	go func() {
 		log.Info("Listening for deprovision messages")
-		for {
-			msg := <-msgBuffer
-			metrics.DeprovisionJobFinished()
-			log.Debug("Processed deprovision message from buffer")
-
-			if msg.Error != "" {
-				// Job failed, mark failure
-				log.Errorf("Deprovision job reporting error: %s", msg.Error)
-				setFailedDeprovisionJob(d.dao, msg)
-				continue
-			}
-
-			instance, err := d.dao.GetServiceInstance(msg.InstanceUUID)
-			if err != nil {
-				log.Errorf(
-					"Error occurred getting service instance [ %s ] after deprovision job:",
-					msg.InstanceUUID,
-				)
-				setFailedDeprovisionJob(d.dao, msg)
-				continue
+		for msg := range msgBuffer {
+			log.Debug("received deprovision message from buffer")
+			if msg.State.State == apb.StateSucceeded {
+				instance, err := d.dao.GetServiceInstance(msg.InstanceUUID)
+				if err != nil {
+					log.Errorf(
+						"Error occurred getting service instance [ %s ] after deprovision job:",
+						msg.InstanceUUID,
+					)
+					setFailedDeprovisionJob(d.dao, msg)
+					continue
+				}
+				if err := cleanupDeprovision(instance, d.dao); err != nil {
+					log.Errorf("Failed cleaning up deprovision after job, error: %v", err)
+					// Cleanup is reporting something has gone wrong. Deprovision overall
+					// has not completed. Mark the job as failed.
+					setFailedDeprovisionJob(d.dao, msg)
+					continue
+				}
 			}
-
-			// Job is not reporting error, cleanup after deprovision
-			err = cleanupDeprovision(instance, d.dao)
-			if err != nil {
-				log.Error("Failed cleaning up deprovision after job, error: %s", err.Error())
-				// Cleanup is reporting something has gone wrong. Deprovision overall
-				// has not completed. Mark the job as failed.
-				setFailedDeprovisionJob(d.dao, msg)
-				continue
-			}
-
-			// No errors reported, deprovision action successfully performed and
-			// broker has successfully cleaned up. Mark depro success
-			if err := d.dao.SetState(msg.InstanceUUID, apb.JobState{
-				Token:   msg.JobToken,
-				State:   apb.StateSucceeded,
-				Podname: msg.PodName,
-				Method:  apb.JobMethodDeprovision,
-			}); err != nil {
-				log.Errorf("failed to set state after deprovision %#v", err)
+			if err := d.dao.SetState(msg.InstanceUUID, msg.State); err != nil {
+				log.Errorf("failed to set state after deprovision %v", err)
 			}
 		}
 	}()
 }
 
 func setFailedDeprovisionJob(dao *dao.Dao, dmsg JobMsg) {
-	if err := dao.SetState(dmsg.InstanceUUID, apb.JobState{
-		Token:   dmsg.JobToken,
-		State:   apb.StateFailed,
-		Podname: dmsg.PodName,
-		Method:  apb.JobMethodDeprovision,
-		Error:   dmsg.Error,
-	}); err != nil {
-		log.Errorf("failed to set state after deprovision %#v", err)
+	// have to set the state here manually as the logic that triggers this is in the subscriber
+	dmsg.State.State = apb.StateFailed
```

Comment:
The stutter is definitely weird. If we don't want to add the method, then `dmsg.JobState.State` might work.

Reply:
Yeah thought about doing that also, but see above comment on perhaps removing the ```JobMsg``` and just having ```JobState.State```